### ISSUE TITLE
Does anyone use with Webpack 3?

### ISSUE THREAD

**Issue by morewry** [Autodesk/theo-loader#110]
Has anyone tried using theo-loader with Webpack 3? If so, did it work? And if so, any tips?

I've tried translating the older config in the README based on migration guides and the CLI command I tested Theo with, but am getting an error. Not sure if I'm doing it wrong or if it's just not feasible with this loader being as outdated with Webpack as it is (https://github.com/Autodesk/theo-loader/pull/101).

```js
{
  test: /tokens.+?\.json$/,
  exclude: shouldExclude,
  use: {
    loader: 'theo-loader'
  }
}
```

```js
new webpack.LoaderOptionsPlugin({
  options: {
    theo: {
      transform: {
        type: 'web'
      },
      format: {
        type: [
          'scss',
          'cssmodules.css',
          'custom-properties.css',
          'module.js',
          'common.js',
          'json'
        ]
      }
    }
  }
})
```

```shell
Version: webpack 3.7.1
Time: 332ms
 Asset     Size  Chunks             Chunk Names
web.js  3.56 kB       0  [emitted]  entry
   [0] ./atokens.json 717 bytes {0} [built] [failed] [1 error]

ERROR in ./atokens.json
Module build failed: Error: Format (...see below...) is not registered
    at transform.chain.fold.e (/Users/rmoore/Projects/buffet/node_modules/theo/lib/index.js:118:30)
    at Either.Left.fold (/Users/rmoore/Projects/buffet/node_modules/data.either/lib/either.js:379:10)
    at convertSync (/Users/rmoore/Projects/buffet/node_modules/theo/lib/index.js:117:6)
    at /Users/rmoore/Projects/buffet/node_modules/data.either/lib/either.js:141:26
    at Object.convert (/Users/rmoore/Projects/buffet/node_modules/theo/lib/index.js:123:26)
    at Object.theoLoader (/Users/rmoore/Projects/buffet/node_modules/theo-loader/dist/index.js:141:20)
```

Using an array I get:

```
Module build failed: Error: Format "List [ "scss", "cssmodules.css", "custom-properties.css", "module.js", "common.js", "json" ]" is not registered
```

I thought maybe it needed to be a string list, but joining the array I get:

```
Module build failed: Error: Format "scss,cssmodules.css,custom-properties.css,module.js,common.js,json" is not registered
```

**Comment by morewry** [comment]
OTOH this works:

```js
new webpack.LoaderOptionsPlugin({
  options: {
    theo: {
      transform: {
        type: 'web'
      },
      format: {
        type: [
          'scss'
        ].join()
      }
    }
  }
})
```

I do have to `.join()` for it to work, and as soon as I try to add more than one format type I begin getting the error. Thoughts?

**Comment by beauroberts** [comment]
@morewry Sorry, haven't updated dependencies in a *very* long time. I just did that and released `4.0.0`!

You need to pass the format name (e.g., `"scss"`, `"common.js"`, as a single string) to `format.type` - it won't accept an Array of values. If you want to be able to import tokens in other formats depending on the context, there are a couple of different approaches:

- Add query parameters to your import statement inline:

```jsx
// Import as a SCSS string
import colorTokens from 'theo-loader?{"transform":{"type":"web"},"format":{"type":"scss"}!./colors.tokens.json';
console.log(colorTokens)
// "$color-background: rgb(244, 246, 249);\n$color-background-alt: rgb(255, 255, 255);"
```

- Add query parameters to your loader configuration in `webpack.config.js`:

```jsx
{
  test: /\.tokens\.json$/,
  loader: 'theo-loader',
  issuer: /\.jsx?$/,
  options: {
    transform: {
        type: 'web',
    },
    format: {
        type: 'common.js',
    },
  },
}
```

In the above case, tokens will be imported as a javascript object in all files with a `.js`/`.jsx` suffix. 

[Check out this part of the docs](https://github.com/Autodesk/theo-loader#formats-and-transforms) and let me know what's confusing!

**Comment by morewry** [comment]
Thanks! That helped! I was looking at this from two angles: one to consume design tokens in an app, one to publish design token modules, which might include pre-building into certain formats.  Looks like in this case webpack is good for the first case, but for the second the Theo CLI has the best results.
